docs/sorting_policy.md

```diff
@@ -0,0 +1,129 @@
+ -->
+
+# YuniKorn Sorting Policies
+The scheduler uses policies allow changing the scheduling behaviour without code changes.
+Policies can be set for:
+* [Applications](#application-sorting)
+* [Nodes](#node-sorting)
+* [Requests](#request-sorting)
+
+## Application sorting
+The application sorting policy is set for each queue via the config.
+A sorting policy setting is only effective on a `leaf` queue.
+Each `leaf` queue can use a different policy.
+
+A `parent` queue will always use the fair policy to sort the child queues.
+
+The following configuration entry sets the application sorting policy to `fifo` for the queue `root.sandbox`: 
+```yaml
+partitions:
+  - name: default
+    queues:
+    - name: root
+      queues:
+      - name: sandbox
+        properties:
+          application.sort.policy: fifo
+```
+
+The only applications that are considered during scheduling must have outstanding requests.
+A filter is applied _while_ sorting the applications to remove all that do not have outstanding requests.
+
+### FifoSortPolicy
+Short description: first in first out, based on application create time  
+Config value: fifo (default)  
+Behaviour:  
+The applications are sorted based on the application create time stamp.
+Since applications can only be added while the system is locked there can never be two applications with the exact same time stamp. 
+
+Before sorting the applications are filtered and must have outstanding requests.
+
+### FairSortPolicy
+Short description: fair based on usage  
+Config value: fair  
+Behaviour:  
+The applications are sorted based on the application usage.
+The usage of the application is defined as all confirmed and unconfirmed allocations for the applications. 
+All resources defined on the application will be taken into account when calculating the usage.
+
+Before sorting the applications are filtered and must have outstanding requests.
+
+### StateAwarePolicy
+Short description: limit of one (1) application in Starting or Accepted state  
+Config value: stateaware  
+Behaviour:  
+The applications are sorted based on the application create time stamp, after filtering based on status.
+Applications states are described in the [application states](object_states.md#application-state) documentation.
+
+Before sorting applications the following filter is applied to the applications:
+* all applications in the state _running_
+* only one of the following two:  
+    * one application in the _starting_ state
+    * one application in the _accepted_ state
+* applications must have outstanding requests
+
+The application in the _accepted_ state is only added if there is no application in the _starting_ state.
+The application in the _starting_ state does not have to have outstanding requests.
+Any application in the _starting_ state will prevent _accepted_ applications from being added to the filtered list.
+
+After the list is filtered all applications are sorted on create time.
+
```

Comment:
It would be nice to give an example to illustrate how this works, e.g with Spark jobs.